src/pages/Home/index.tsx

```diff
@@ -52,19 +59,25 @@ const Homepage = ({ order, setOrder }: OrderProps): JSX.Element => {
   };
 
   useEffect(() => {
+    if(data){
+      setFilteredBobaList(data)
+    }
     if (searchQuery) {
-      const filterlist = filterBobaList(bobaList, searchQuery);
+      const filterlist = filterBobaList(data, searchQuery);
       if (filterlist.length === 0) {
         setFilteredBobaList([]);
       } else {
         setFilteredBobaList(filterlist);
       }
-    } else {
-      setFilteredBobaList(bobaList);
     }
-  }, [searchQuery]);
+  }, [searchQuery,data]);
 
-  // console.log(order, 'this is order3'); // this is for you, u said you want to see the console.log
+  if (loading)
+    return (
+      <Spinner animation='border' role='status'>
+        <span className='visually-hidden'>Loading...</span>
+      </Spinner>
+    );
```

Comment:
Put your loading Spinner logic right at line 98 return
```
return loading
   ? <Spinner />
   : <Container />

Reply:
This ternary is generic but you can and should use React's Suspense
https://react.dev/reference/react/Suspense

Reply:
i try using react suspense, i think it render too quick that i didnt see anything like loading on the screen .

Reply:
tried but didnt work so closing this